### PR TITLE
Fiid 201/adecuadacion del modal de rentabilidad

### PR DESCRIPTION
**Pull request by francoFinniu**
- "adecuadacion del modal de rentabilidad para la calculadora" [FIID-201](https://finniu-tech.atlassian.net/browse/FIID-201)
- change investment_sumilation and add : simulation loading , success and erro modal.

Screenshot 2024-07-24 122011
Screenshot 2024-07-24 122027
Screenshot 2024-07-24 122046
Screenshot 2024-07-24 122105

### CODE REVIEW

File: lib/presentation/screens/catalog/widgets/investment_simulation.dart [hunk 1]
```diff
@@ -107,112 +107,65 @@ class BodySimulation extends ConsumerWidget {
...
+      currency: isSoles ? 'nuevo sol' : 'dolar',
+    );
+
+    final response =
+        ref.watch(calculateInvestmentFutureProvider(calculatorInput));
+
+    return response.when(
+      data: (data) {
+        return SimulationSuccess(
+          mouthInvestment: widget.mouthInvestment,
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/investment_simulation.dart:145]: mouth ? or month ?

File: lib/presentation/screens/catalog/widgets/simulation_modal/simulation_loading.dart [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import 'package:finniu/presentation/screens/catalog/widgets/send_proof_button.dart';
+import 'package:finniu/presentation/screens/catalog/widgets/text_poppins.dart';
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+class SimulationLoading extends ConsumerWidget {
+  const SimulationLoading({
+    super.key,
+    required this.startingAmount,
+    required this.mouthInvestment,
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/simulation_modal/simulation_loading.dart:13]: mouth?

**Review by harolCalzada — CHANGES_REQUESTED** [review]

File: lib/presentation/screens/catalog/widgets/simulation_modal/simulation_loading.dart (hunk 2 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/simulation_modal/simulation_loading.dart]: change to month

File: lib/presentation/screens/catalog/widgets/investment_simulation.dart (hunk 1 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/investment_simulation.dart]: change to month

**Review by harolCalzada — APPROVED** [review]

**Pull request merged**